src/main/java/net/rptools/maptool/client/functions/TemplateFunctions.java

```diff
@@ -0,0 +1,782 @@
+
+  /** The minimum radius value allowed. */
+  private final int MIN_RADIUS = AbstractTemplate.MIN_RADIUS;
+
+  private TemplateFunctions() {
+    super(
+        0,
+        5,
+        FUNC_CREATE_TEMPLATE,
+        FUNC_GET_TEMPLATE_RADIUS,
+        FUNC_GET_TEMPLATE_X,
+        FUNC_GET_TEMPLATE_Y,
+        FUNC_MOVE_TEMPLATE);
+  }
+
+  /**
+   * TemplateType Enumeration, including:
+   *
+   * <ul>
+   *   <li><code>alias</code> property for matching template type by an alias
+   *   <li><code>template</code> method to return the appropriate template class
+   */
+  private enum TemplateType {
+    BlastTemplate("Blast") {
+      @Override
+      public AbstractTemplate template() {
+        return new BlastTemplate();
+      }
+    },
+    BurstTemplate("Burst") {
+      @Override
+      public AbstractTemplate template() {
+        return new BurstTemplate();
+      }
+    },
+    ConeTemplate("Cone") {
+      @Override
+      public AbstractTemplate template() {
+        return new ConeTemplate();
+      }
+    },
+    LineTemplate("Line") {
+      @Override
+      public AbstractTemplate template() {
+        return new LineTemplate();
+      }
+    },
+    LineCellTemplate("LineCell") {
+      @Override
+      public AbstractTemplate template() {
+        return new LineCellTemplate();
+      }
+    },
+    RadiusTemplate("Radius") {
+      @Override
+      public AbstractTemplate template() {
+        return new RadiusTemplate();
+      }
+    },
+    RadiusCellTemplate("RadiusCell") {
+      @Override
+      public AbstractTemplate template() {
+        return new RadiusCellTemplate();
+      }
+    },
+    WallTemplate("Wall") {
+      @Override
+      public AbstractTemplate template() {
+        return new WallTemplate();
+      }
+    };
+
+    TemplateType(String alias) {
+      this.alias = alias;
+    }
+
+    public AbstractTemplate template() {
+      return null;
+    }
+
+    private final String alias;
+
+    public String getAlias() {
+      return alias;
+    }
+
+    public static TemplateType valueOfAlias(String alias) {
+      for (TemplateType tt : values()) {
+        if (tt.alias.equalsIgnoreCase(alias)) {
+          return tt;
+        }
+      }
+      return null;
+    }
+  }
+
+  /** Evaluate the function(s) */
+  @Override
+  public Object childEvaluate(
+      Parser parser, VariableResolver resolver, String functionName, List<Object> parameters)
+      throws ParserException {
+    FunctionUtil.blockUntrustedMacro(functionName);
+
+    if (FUNC_CREATE_TEMPLATE.equalsIgnoreCase(functionName)) {
+      // templateType, radius/wallPath [, units, [, options [, delim [, mapRef ]]]]
+      FunctionUtil.checkNumberParam(FUNC_CREATE_TEMPLATE, parameters, 2, 6);
+
+      // template type, determine first by alias, then name (i.e. enum value)
+      String templateType = parameters.getFirst().toString().toUpperCase().trim();
+      TemplateType tt = TemplateType.valueOfAlias(templateType);
+      if (tt == null) {
```

Comment:
I think we can just use the abbreviated names in `createTemplate()` and not bother supporting the `…Template` alternatives.